src/service-override/views.ts

```diff
@@ -388,12 +388,16 @@ interface CustomViewOption {
     icon?: keyof typeof Codicon
     render?(container: HTMLElement): void
   }[]
+  viewContainer?: ViewContainer
+  canToggleVisibility?: boolean
+  hideByDefault?: boolean
+  collapsed?: boolean
 }
 
 function registerCustomView (options: CustomViewOption): IDisposable {
   const iconUrl = options.icon != null ? URI.parse(options.icon) : undefined
 
-  const VIEW_CONTAINER = Registry.as<IViewContainersRegistry>(ViewExtensions.ViewContainersRegistry).registerViewContainer({
+  const VIEW_CONTAINER = options.viewContainer ?? Registry.as<IViewContainersRegistry>(ViewExtensions.ViewContainersRegistry).registerViewContainer({
```

Comment:
How did you create the view container?

Reply:
btw, you may want to NOT deregistering the view container then line 456

Reply:
Good one. I get them through `IViewDescriptorService`. I run this:

```
viewDescriptorService
            .getViewContainersByLocation(0)
```

Where 0 corresponds to the sidebar.

Reply:
Ok I have no idea why the variable is in upper case, do you mind changing it?

Reply:
Sounds good, it's update!